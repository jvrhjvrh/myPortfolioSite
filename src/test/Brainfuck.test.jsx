import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Brainfuck from '../pages/brainfuck';

const hardcoreCode = '+[------->++<]>--.-------.[--->+<]>---.+[--->+<]>.-[->+++<]>+.+[---->+<]>+++.+[----->+<]>.------------.++++++++++.------.--[--->+<]>-.---[->++++<]>.------------.+.++++++++++.+[---->+<]>+++.---[->++++<]>-.++.-----.-----------.+++++++++++++.[-->+++++<]>+++.-[--->++<]>--.-------.--[--->+<]>---.--------------.-[--->+<]>-.--[->++++<]>+.--[->+++<]>.----.+++++++.--[--->+<]>--.--[->++++<]>+.--[->+++<]>.----.+++++++.--[--->+<]>--.--[->++++<]>+.--[->+++<]>.----.+++++++.---[->+++<]>-.-------------.-[->++<]>.[-->+<]>+++.>+[--->++<]>+.---[----->+<]>.[->+++++<]>-.---[->++++<]>+.+[-->+<]>++.---[->++<]>.[-->+<]>++++.+.++++.+.--.-----.+++++++.[-->+++<]>+.-[--->++<]>.-------.+++++++.-.+.--.++.>++++++++++.-[----->+<]>-.-.-.++.++++.+.---------.++++++++.--------.------------.++++.+..+.------.----[->++<]>.-[--->++<]>--.+.+.-------.++++++.+.---------.++++++++.+[-->+++<]>+.[-->+<]>+++.++++++.---[-->+++<]>.-.--------.+++.+++++.------.--.+.++.---.';
const brackets = '[[[[]]]]';

it('renders correctly', () => {
    const component = render(<div data-testid="custom-element"><Brainfuck /></div>);
    expect(component.getByTestId('custom-element')).toMatchSnapshot();
});

test('should show Hello World!', () => {
    const component = render(<Brainfuck />);
    const input = component.getByPlaceholderText('Insert brainfuck code here');
    fireEvent.change(input, { target: { value: '++++++++[>++++[>++>+++>+++>+<<<<-]>+>+>->>+[<]<-]>>.>---.+++++++..+++.>>.<-.<.+++.------.--------.>>+.>++.' } });
    const button = component.getByText('Translate ->');
    fireEvent.click(button);
    component.getByText('Hello World!');
});

test('should only get .+-><[]', () => {
    const component = render(<Brainfuck />);
    const input = component.getByPlaceholderText('Insert brainfuck code here');
    fireEvent.change(input, { target: { value: '..++++dsad' } });
    component.getByText('..++++');
});

test('more complicated brainfuck code', () => {
    const component = render(<Brainfuck />);
    const input = component.getByPlaceholderText('Insert brainfuck code here');
    fireEvent.change(input, { target: { value: hardcoreCode } });
    const button = component.getByText('Translate ->');
    fireEvent.click(button);
});

test('brackets', () => {
    const component = render(<Brainfuck />);
    const input = component.getByPlaceholderText('Insert brainfuck code here');
    fireEvent.change(input, { target: { value: brackets } });
    const button = component.getByText('Translate ->');
    fireEvent.click(button);
});

test('dont go into negative', () => {
    const component = render(<Brainfuck />);
    const input = component.getByPlaceholderText('Insert brainfuck code here');
    fireEvent.change(input, { target: { value: '<' } });
    const button = component.getByText('Translate ->');
    fireEvent.click(button);
});
