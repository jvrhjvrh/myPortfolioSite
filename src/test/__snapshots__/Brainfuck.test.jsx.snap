// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-testid="custom-element"
>
  <div
    class="app"
    data-testid="brainfuck-page"
  >
    <div
      class="ant-card ant-card-bordered"
    >
      <div
        class="ant-card-head"
        style="text-align: center;"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Code
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
        style="height: 100%;"
      >
        <div
          class="text-container"
        >
          <textarea
            class="ant-input textarea-input"
            placeholder="Insert brainfuck code here"
            style="height: 0px; min-height: -40px; max-height: 9007199254740991px; overflow-y: hidden; overflow-x: hidden;"
          />
        </div>
      </div>
    </div>
    <button
      class="ant-btn translate-button ant-btn-primary"
      type="button"
    >
      <span>
        Translate -&gt;
      </span>
    </button>
    <div
      class="ant-card ant-card-bordered"
    >
      <div
        class="ant-card-head"
        style="text-align: center;"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Translation
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
        style="height: 100%;"
      >
        <div
          class="text-container"
        >
          <textarea
            class="ant-input textarea-input ant-input-disabled"
            disabled=""
            placeholder="Translation here"
            style="height: 0px; min-height: -40px; max-height: 9007199254740991px; overflow-y: hidden; overflow-x: hidden;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
